From d35bd7cf635913f6e4d8dc3ec8fdaf7ea166d313 Mon Sep 17 00:00:00 2001
From: Alaa Hleihel <ahleihel@redhat.com>
Date: Tue, 12 May 2020 10:53:38 -0400
Subject: [PATCH 132/312] [netdrv] net/mlx5: Reduce No CQ found log level from
 warn to debug

Message-id: <20200512105530.4207-13-ahleihel@redhat.com>
Patchwork-id: 306884
Patchwork-instance: patchwork
O-Subject: [RHEL8.3 BZ 1789382 012/124] net/mlx5: Reduce No CQ found log level from warn to debug
Bugzilla: 1789382
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Kamal Heib <kheib@redhat.com>
RH-Acked-by: Jarod Wilson <jarod@redhat.com>

Bugzilla: http://bugzilla.redhat.com/1789382
Upstream: v5.6-rc1

commit 7396ae3d1cfe05be42e5f6b1883c9ed594cc42ba
Author: Parav Pandit <parav@mellanox.com>
Date:   Thu Dec 5 16:06:20 2019 -0600

    net/mlx5: Reduce No CQ found log level from warn to debug

    In below sequence, a EQE entry arrives for a CQ which is on the path of
    being destroyed.

               cpu-0               cpu-1
               ------              -----
    mlx5_core_destroy_cq()      mlx5_eq_comp_int()
      mlx5_eq_del_cq()          [..]
        radix_tree_delete()     [..]
      [..]                         mlx5_eq_cq_get() /* Didn't find CQ is
                                                     * a valid case.
                                                     */
      /* destroy CQ in hw */
      mlx5_cmd_exec()

    This is still a valid scenario and correct delete CQ sequence, as
    mirror of the CQ create sequence.
    Hence, suppress the non harmful debug message from warn to debug level.
    Keep the debug log message rate limited because user application can
    trigger it repeatedly.

    Signed-off-by: Parav Pandit <parav@mellanox.com>
    Signed-off-by: Saeed Mahameed <saeedm@mellanox.com>

Signed-off-by: Alaa Hleihel <ahleihel@redhat.com>
Signed-off-by: Frantisek Hrbata <fhrbata@redhat.com>
---
 drivers/net/ethernet/mellanox/mlx5/core/eq.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/mellanox/mlx5/core/eq.c b/drivers/net/ethernet/mellanox/mlx5/core/eq.c
index 580c71cb9dfa..2c716abc0f27 100644
--- a/drivers/net/ethernet/mellanox/mlx5/core/eq.c
+++ b/drivers/net/ethernet/mellanox/mlx5/core/eq.c
@@ -156,7 +156,8 @@ static int mlx5_eq_comp_int(struct notifier_block *nb,
 			cq->comp(cq, eqe);
 			mlx5_cq_put(cq);
 		} else {
-			mlx5_core_warn(eq->dev, "Completion event for bogus CQ 0x%x\n", cqn);
+			dev_dbg_ratelimited(eq->dev->device,
+					    "Completion event for bogus CQ 0x%x\n", cqn);
 		}
 
 		++eq->cons_index;
-- 
2.13.6

